content/references/kata-trainer.md

```diff
@@ -3,41 +3,78 @@ title: Kata Trainer
 kind: reference
 ---
 
-:::note
-*This page is a stub, please consider contributing to it.*
-:::
+import ThemedImage from "@theme/ThemedImage";
+
+The kata trainer is the interface you're presented when you click "Train" on a kata. It is here that you write your solution, run tests and attempt to complete the kata. Since this is where you'll be spending the majority of your time, it is recommended that you familiarize yourself with the interface and its functionality. That way, you can focus entirely on solving the kata.
 
-The kata trainer is the interface you use for completing kata. 
+<ThemedImage
+  alt="Kata trainer interface"
+  sources={{
+    light: require("./img/kata-trainer-light.png").default,
+    dark: require("./img/kata-trainer-dark.png").default,
+  }}
+/>
+*The interface: 1) View Panel 2) Language selection 3) Language version 4) Solution Editor 5) Sample Tests Editor 6) Action buttons 7) Enable VIM/EMACS mode*
 
-![Kata Trainer](./img/kata-trainer.png)
+## View panel
+On the left side of the interface you will find two views split into separate tabs:
+- **Instructions**: The kata description with information about the task.
+- **Output**: A view showing the test results as well as anything you choose to print to `stdout` from within your solution code.
+- **Past Solutions**: If you have completed this kata before (possibly in a different language), you can view your past solutions here. If this is your first attempt at solving the kata, this tab will not be displayed.
+
+You can switch freely between these tabs at any time during your training on the kata. Note that running tests will automatically display the Output panel.
 
 ## Editors
 
+The right half of the interface is dedicated to writing and testing your solution. At the very top you'll find the following options:
+- **Language**: Here you select the language you wish to train in. Many kata feature translations into multiple languages, and you can train in any of them by selecting the appropriate language here. If your training setup includes multiple languages, make sure the correct one is selected before you begin writing any code.
+- **Language version**: Over time, programming languages evolve and new versions are released, which may then get added to Codewars. Select your preferred language version from the available options here. Note that your choice affects both your solution and the test suite and can have a significant impact on performance under certain circumstances.
+- **VIM/EMACS mode**: If you prefer your editor to replicate a VIM or EMACS experience, you can toggle those modes with the buttons on the right. Because the <kbd>Esc</kbd> key has a special function in VIM mode, its usual behavior of toggling full-screen mode is disabled.
```

Comment:
```suggestion
- **VIM/EMACS mode**: If you prefer your editor to replicate a VIM or EMACS experience, you can toggle those modes with the buttons on the right.
```

Note on 'usual behaviour' feels to me a bit out of order, because the usual behaviour were not mentioned yet. Maybe move the note about redefined Esc to the place where it's actually introduced? (see suggestion below)

Reply:
By the way, the default is [sublime](https://codemirror.net/demo/sublime.html).